cranelift/filetests/filetests/isa/aarch64/simd.clif

```diff
@@ -0,0 +1,49 @@
+test compile
+target aarch64
+
+function %f1() -> i64x2 {
+block0:
+  v0 = iconst.i64 281474976710657
+  v1 = splat.i64x2 v0
+  return v1
+}
+
+; check:  stp fp, lr, [sp, #-16]!
+; nextln:  mov fp, sp
+; nextln:  movz x0, #1
+; nextln:  movk x0, #1, LSL #48
+; nextln:  dup v0.2d, x0
+; nextln:  mov sp, fp
+; nextln:  ldp fp, lr, [sp], #16
+; nextln:  ret
+
+function %f2() -> i16x8 {
+block0:
+  v0 = iconst.i32 42679
+  v1 = ireduce.i16 v0
+  v2 = splat.i16x8 v1
+  return v2
+}
+
+; check:  stp fp, lr, [sp, #-16]!
+; nextln:  mov fp, sp
+; nextln:  movz x0, #42679
+; nextln:  dup v0.8h, w0
+; nextln:  mov sp, fp
+; nextln:  ldp fp, lr, [sp], #16
+; nextln:  ret
+
+function %f3() -> b8x16 {
+block0:
+  v0 = bconst.b32 true
+  v1 = breduce.b8 v0
+  v2 = splat.b8x16 v1
+  return v2
+}
+
+; check:  stp fp, lr, [sp, #-16]!
+; nextln:  mov fp, sp
+; nextln:  movi v0.16b, #1
```

Comment:
This *should* be `-1`, as you noted, to be consistent with #2058; after a bit of digging, this seems to be a simple mistake (on my part) in `inst_predicates::is_constant_64bit()`, where the `bool` actually becomes a `u64`, currently `1` or `0`. That should simply be -1 masked down to the appropriate width, I think.